libgloss/arc64/readme-hostlink.md

```diff
@@ -0,0 +1,88 @@
+Metaware hostlink IO
+====================
+
+This directory includes target-side implementation of Metaware hostlink
+interface see Contents section. Target program can use Metaware hostlink
+interface to send messages to nsim simulator or mdb debugger (it can be
+attached to HW or nsim).
+
+Quick start
+-----------
+To link with this version of libgloss please add `-specs=hl.specs` to baremetal
+version of ARC gcc (arc64-elf).
+
+Lets build and run simple program:
+
+    $ cat hello.c
+    #include <stdio.h>
+
+    int main()
+    {
+            printf("Hello World!\n");
+
+            return 0;
+    }
+    $ arc64-elf-gcc -specs=hl.specs ./hello.c -o hello
+    $ nsimdrv -prop=nsim_isa_family=arc64 -prop=nsim_isa_div_rem_option=2 -prop=nsim_isa_mpy_option=3 -prop=nsim_hlink_gnu_io_ext=1 ./hello
+    Hello World!
+
+Where `-prop=nsim_isa_family=arc64 -prop=nsim_isa_div_rem_option=2 -prop=nsim_isa_mpy_option=3`
+is specific to your version of ARC CPU.
+Option `-prop=nsim_hlink_gnu_io_ext=1` enables GNU IO extension for nSIM which
+is used for some system calls.
+
+**NB:** Metaware hostlink requires symbols `__HOSTLINK__` and `_hl_blockedPeek`
+to be present. So stripped binary won't work properly with Metaware hostlink.
+
```

Comment:
a good place to mention `fstat()` and `gettimeofday()` noncompliances.

Reply:
`fstat()` is now mentioned.

`gettimeofday()` is compliant with [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/functions/gettimeofday.html). POSIX says `If tzp is not a null pointer, the behavior is unspecified.`. Evem glibc does the same [thing](https://sourceware.org/git/?p=glibc.git;a=blob;f=time/gettimeofday.c;h=18c84f68c7fcc9bbd34a389ebf63d7e517f43689;hb=HEAD).